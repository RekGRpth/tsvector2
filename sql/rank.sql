CREATE EXTENSION tsvector2;

--ranking
SELECT ts_rank(' a:1 s:2C d g'::tsvector2, 'a | s');
SELECT ts_rank(' a:1 sa:2C d g'::tsvector2, 'a | s');
SELECT ts_rank(' a:1 sa:2C d g'::tsvector2, 'a | s:*');
SELECT ts_rank(' a:1 sa:2C d g'::tsvector2, 'a | sa:*');
SELECT ts_rank(' a:1 s:2B d g'::tsvector2, 'a | s');
SELECT ts_rank(' a:1 s:2 d g'::tsvector2, 'a | s');
SELECT ts_rank(' a:1 s:2C d g'::tsvector2, 'a & s');
SELECT ts_rank(' a:1 s:2B d g'::tsvector2, 'a & s');
SELECT ts_rank(' a:1 s:2 d g'::tsvector2, 'a & s');

SELECT ts_rank_cd(' a:1 s:2C d g'::tsvector2, 'a | s');
SELECT ts_rank_cd(' a:1 sa:2C d g'::tsvector2, 'a | s');
SELECT ts_rank_cd(' a:1 sa:2C d g'::tsvector2, 'a | s:*');
SELECT ts_rank_cd(' a:1 sa:2C d g'::tsvector2, 'a | sa:*');
SELECT ts_rank_cd(' a:1 sa:3C sab:2c d g'::tsvector2, 'a | sa:*');
SELECT ts_rank_cd(' a:1 s:2B d g'::tsvector2, 'a | s');
SELECT ts_rank_cd(' a:1 s:2 d g'::tsvector2, 'a | s');
SELECT ts_rank_cd(' a:1 s:2C d g'::tsvector2, 'a & s');
SELECT ts_rank_cd(' a:1 s:2B d g'::tsvector2, 'a & s');
SELECT ts_rank_cd(' a:1 s:2 d g'::tsvector2, 'a & s');

SELECT ts_rank_cd(' a:1 s:2A d g'::tsvector2, 'a <-> s');
SELECT ts_rank_cd(' a:1 s:2C d g'::tsvector2, 'a <-> s');
SELECT ts_rank_cd(' a:1 s:2 d g'::tsvector2, 'a <-> s');
SELECT ts_rank_cd(' a:1 s:2 d:2A g'::tsvector2, 'a <-> s');
SELECT ts_rank_cd(' a:1 s:2,3A d:2A g'::tsvector2, 'a <2> s:A');
SELECT ts_rank_cd(' a:1 b:2 s:3A d:2A g'::tsvector2, 'a <2> s:A');
SELECT ts_rank_cd(' a:1 sa:2D sb:2A g'::tsvector2, 'a <-> s:*');
SELECT ts_rank_cd(' a:1 sa:2A sb:2D g'::tsvector2, 'a <-> s:*');
SELECT ts_rank_cd(' a:1 sa:2A sb:2D g'::tsvector2, 'a <-> s:* <-> sa:A');
SELECT ts_rank_cd(' a:1 sa:2A sb:2D g'::tsvector2, 'a <-> s:* <-> sa:B');

SELECT ts_rank_cd(to_tsvector2('english', '
Day after day, day after day,
  We stuck, nor breath nor motion,
As idle as a painted Ship
  Upon a painted Ocean.
Water, water, every where
  And all the boards did shrink;
Water, water, every where,
  Nor any drop to drink.
S. T. Coleridge (1772-1834)
'), to_tsquery('english', 'paint&water'));

SELECT ts_rank_cd(to_tsvector2('english', '
Day after day, day after day,
  We stuck, nor breath nor motion,
As idle as a painted Ship
  Upon a painted Ocean.
Water, water, every where
  And all the boards did shrink;
Water, water, every where,
  Nor any drop to drink.
S. T. Coleridge (1772-1834)
'), to_tsquery('english', 'breath&motion&water'));

SELECT ts_rank_cd(to_tsvector2('english', '
Day after day, day after day,
  We stuck, nor breath nor motion,
As idle as a painted Ship
  Upon a painted Ocean.
Water, water, every where
  And all the boards did shrink;
Water, water, every where,
  Nor any drop to drink.
S. T. Coleridge (1772-1834)
'), to_tsquery('english', 'ocean'));

SELECT ts_rank_cd(to_tsvector2('english', '
Day after day, day after day,
  We stuck, nor breath nor motion,
As idle as a painted Ship
  Upon a painted Ocean.
Water, water, every where
  And all the boards did shrink;
Water, water, every where,
  Nor any drop to drink.
S. T. Coleridge (1772-1834)
'), to_tsquery('english', 'painted <-> Ship'));

SELECT ts_rank_cd(strip(to_tsvector2('both stripped')),
                  to_tsquery('both & stripped'));

SELECT ts_rank_cd(to_tsvector2('unstripped') || strip(to_tsvector2('stripped')),
                  to_tsquery('unstripped & stripped'));

DROP EXTENSION tsvector2;
