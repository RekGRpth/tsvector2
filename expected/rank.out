CREATE EXTENSION tsvector2;
--ranking
SELECT ts_rank(' a:1 s:2C d g'::tsvector2, 'a | s');
  ts_rank  
-----------
 0.0911891
(1 row)

SELECT ts_rank(' a:1 sa:2C d g'::tsvector2, 'a | s');
  ts_rank  
-----------
 0.0303964
(1 row)

SELECT ts_rank(' a:1 sa:2C d g'::tsvector2, 'a | s:*');
  ts_rank  
-----------
 0.0911891
(1 row)

SELECT ts_rank(' a:1 sa:2C d g'::tsvector2, 'a | sa:*');
  ts_rank  
-----------
 0.0911891
(1 row)

SELECT ts_rank(' a:1 s:2B d g'::tsvector2, 'a | s');
 ts_rank  
----------
 0.151982
(1 row)

SELECT ts_rank(' a:1 s:2 d g'::tsvector2, 'a | s');
  ts_rank  
-----------
 0.0607927
(1 row)

SELECT ts_rank(' a:1 s:2C d g'::tsvector2, 'a & s');
 ts_rank  
----------
 0.140153
(1 row)

SELECT ts_rank(' a:1 s:2B d g'::tsvector2, 'a & s');
 ts_rank  
----------
 0.198206
(1 row)

SELECT ts_rank(' a:1 s:2 d g'::tsvector2, 'a & s');
  ts_rank  
-----------
 0.0991032
(1 row)

SELECT ts_rank_cd(' a:1 s:2C d g'::tsvector2, 'a | s');
 ts_rank_cd 
------------
        0.3
(1 row)

SELECT ts_rank_cd(' a:1 sa:2C d g'::tsvector2, 'a | s');
 ts_rank_cd 
------------
        0.1
(1 row)

SELECT ts_rank_cd(' a:1 sa:2C d g'::tsvector2, 'a | s:*');
 ts_rank_cd 
------------
        0.3
(1 row)

SELECT ts_rank_cd(' a:1 sa:2C d g'::tsvector2, 'a | sa:*');
 ts_rank_cd 
------------
        0.3
(1 row)

SELECT ts_rank_cd(' a:1 sa:3C sab:2c d g'::tsvector2, 'a | sa:*');
 ts_rank_cd 
------------
        0.5
(1 row)

SELECT ts_rank_cd(' a:1 s:2B d g'::tsvector2, 'a | s');
 ts_rank_cd 
------------
        0.5
(1 row)

SELECT ts_rank_cd(' a:1 s:2 d g'::tsvector2, 'a | s');
 ts_rank_cd 
------------
        0.2
(1 row)

SELECT ts_rank_cd(' a:1 s:2C d g'::tsvector2, 'a & s');
 ts_rank_cd 
------------
   0.133333
(1 row)

SELECT ts_rank_cd(' a:1 s:2B d g'::tsvector2, 'a & s');
 ts_rank_cd 
------------
       0.16
(1 row)

SELECT ts_rank_cd(' a:1 s:2 d g'::tsvector2, 'a & s');
 ts_rank_cd 
------------
        0.1
(1 row)

SELECT ts_rank_cd(' a:1 s:2A d g'::tsvector2, 'a <-> s');
 ts_rank_cd 
------------
   0.181818
(1 row)

SELECT ts_rank_cd(' a:1 s:2C d g'::tsvector2, 'a <-> s');
 ts_rank_cd 
------------
   0.133333
(1 row)

SELECT ts_rank_cd(' a:1 s:2 d g'::tsvector2, 'a <-> s');
 ts_rank_cd 
------------
        0.1
(1 row)

SELECT ts_rank_cd(' a:1 s:2 d:2A g'::tsvector2, 'a <-> s');
 ts_rank_cd 
------------
        0.1
(1 row)

SELECT ts_rank_cd(' a:1 s:2,3A d:2A g'::tsvector2, 'a <2> s:A');
 ts_rank_cd 
------------
  0.0909091
(1 row)

SELECT ts_rank_cd(' a:1 b:2 s:3A d:2A g'::tsvector2, 'a <2> s:A');
 ts_rank_cd 
------------
  0.0909091
(1 row)

SELECT ts_rank_cd(' a:1 sa:2D sb:2A g'::tsvector2, 'a <-> s:*');
 ts_rank_cd 
------------
        0.1
(1 row)

SELECT ts_rank_cd(' a:1 sa:2A sb:2D g'::tsvector2, 'a <-> s:*');
 ts_rank_cd 
------------
        0.1
(1 row)

SELECT ts_rank_cd(' a:1 sa:2A sb:2D g'::tsvector2, 'a <-> s:* <-> sa:A');
 ts_rank_cd 
------------
          0
(1 row)

SELECT ts_rank_cd(' a:1 sa:2A sb:2D g'::tsvector2, 'a <-> s:* <-> sa:B');
 ts_rank_cd 
------------
          0
(1 row)

SELECT ts_rank_cd(to_tsvector2('english', '
Day after day, day after day,
  We stuck, nor breath nor motion,
As idle as a painted Ship
  Upon a painted Ocean.
Water, water, every where
  And all the boards did shrink;
Water, water, every where,
  Nor any drop to drink.
S. T. Coleridge (1772-1834)
'), to_tsquery('english', 'paint&water'));
 ts_rank_cd 
------------
       0.05
(1 row)

SELECT ts_rank_cd(to_tsvector2('english', '
Day after day, day after day,
  We stuck, nor breath nor motion,
As idle as a painted Ship
  Upon a painted Ocean.
Water, water, every where
  And all the boards did shrink;
Water, water, every where,
  Nor any drop to drink.
S. T. Coleridge (1772-1834)
'), to_tsquery('english', 'breath&motion&water'));
 ts_rank_cd 
------------
 0.00833333
(1 row)

SELECT ts_rank_cd(to_tsvector2('english', '
Day after day, day after day,
  We stuck, nor breath nor motion,
As idle as a painted Ship
  Upon a painted Ocean.
Water, water, every where
  And all the boards did shrink;
Water, water, every where,
  Nor any drop to drink.
S. T. Coleridge (1772-1834)
'), to_tsquery('english', 'ocean'));
 ts_rank_cd 
------------
        0.1
(1 row)

SELECT ts_rank_cd(to_tsvector2('english', '
Day after day, day after day,
  We stuck, nor breath nor motion,
As idle as a painted Ship
  Upon a painted Ocean.
Water, water, every where
  And all the boards did shrink;
Water, water, every where,
  Nor any drop to drink.
S. T. Coleridge (1772-1834)
'), to_tsquery('english', 'painted <-> Ship'));
 ts_rank_cd 
------------
        0.1
(1 row)

SELECT ts_rank_cd(strip(to_tsvector2('both stripped')),
                  to_tsquery('both & stripped'));
 ts_rank_cd 
------------
          0
(1 row)

SELECT ts_rank_cd(to_tsvector2('unstripped') || strip(to_tsvector2('stripped')),
                  to_tsquery('unstripped & stripped'));
 ts_rank_cd 
------------
          0
(1 row)

DROP EXTENSION tsvector2;
